zerver/actions/users.py

```diff
@@ -826,6 +842,41 @@ def do_update_bot_config_data(bot_profile: UserProfile, config_data: dict[str, s
     )
 
 
+@transaction.atomic(durable=True)
+def do_update_bot_type(
+    bot_profile: UserProfile, bot_type: int, *, acting_user: UserProfile | None
+) -> None:
+    previous_bot_type = bot_profile.bot_type
+
+    bot_profile.bot_type = bot_type
+    bot_profile.save(update_fields=["bot_type"])
```

Comment:
Do we have any caches of bots that may need special flushing logic? I'd expect no, but `git grep cache_with_key` should help audit.